### DISCUSSION TITLE
Table: Sort value mapped column by display text, not by value

### DISCUSSION THREAD

**Discussion by a-mazed** [grafana/grafana#43280] [Dashboards, Panels & Visualizations]
**What happened**:
When using value mapping for one of a column in my table, I want to sort this column by display text. But it can be sorted only by source values.
For example

```
value | Display text
--------------------
1     |  B
2     |  A
3     |  C
```
In this case, I never get text ordered as ABC or CBA, only BAC or CAB

**What you expected to happen**:
I want to sort data in column by display text instead of values

**Environment**:
- Grafana version: 8.3.3
- Data source type & version: SQL
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by Elfo404** [comment]
Moving this to a discussion as it seems more like a feature request for the current table panel.

**Reply by liubofeng** [reply]
Is there a solution to this problem now? @a-mazed @Elfo404  Thank you~

**Comment by dmgeurts** [comment]
I've just encountered the same issue. Network device port listing, and applied the following value mapping to be able to sort them sequentially: ^eth([0-9]?)/([1-9])$ >> eth$1/0$2

And found that the sort happens before the mapping, not after as I had expected.

**Comment by xkilian** [comment]
Having diffrent types of sorts should be available for the table panel to account for lists of:

- text
- numbers (1,2,3, .. 11,12,13)
- mixed eth1/1, eth1/11, eth1/12

@dmgeurts cool workaround for basic use-cases, too bad it does not work in the current table.

**Comment by jars99** [comment]
I have a table where 1 maps to Yes and 0 or null map to No.  I can't get all the "No" values to sort together because the table is sorting before the mapped value.  It would be helpful to have an override where I could toggle between sort by "value" or "mapped value".

**Comment by timolux** [comment]
I have the same problem with a column consisting of URLs. I'm using a regular expression in a field override for value mapping. It removes the leading http[s]:// and everything after the next slash (after the TLD). The table shows the values correctly, but when I sort the column, it will be sorted according to the values before the override. In this case, everything which started with "http://" will be shown before (or after, depending on direction) any value that started with "https://" before the override.

Grafana Version is 9.4.7, with Prometheus as the Data Source

**Comment by precisionconage** [comment]
This [epic](https://github.com/grafana/grafana/issues/73237) has an "Allow value mappings to happen before a transformation" feature request, which would solve this issue, but it's apparently closed for further discussion. Hopefully there's an update on it soon.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #83013. Feel free to continue the discussion around this there. Thank you!
